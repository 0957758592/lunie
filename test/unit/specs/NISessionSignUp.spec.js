import Vuex from 'vuex'
import Vuelidate from 'vuelidate'
import { mount, createLocalVue } from 'vue-test-utils'
import htmlBeautify from 'html-beautify'
import NISessionSignUp from 'common/NiSessionSignUp'

const user = require('renderer/vuex/modules/user').default({})
const notifications = require('renderer/vuex/modules/notifications').default({})

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuelidate)

describe('NISessionSignUp', () => {
  let wrapper, store

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        user,
        notifications
      },
      actions: {
        async createKey () {
          return {}
        }
      }
    })
    wrapper = mount(NISessionSignUp, {
      localVue,
      store
    })
    store.commit = jest.fn()
    store.dispatch = jest.fn(() => Promise.resolve({}))
  })

  it('has the expected html structure', () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })

  it('should go back to the welcome screen on click', () => {
    wrapper.findAll('.ni-session-header a').at(0).trigger('click')
    expect(store.commit.mock.calls[0][0]).toBe('setModalSessionState')
    expect(store.commit.mock.calls[0][1]).toBe('welcome')
  })

  it('should open the help model on click', () => {
    wrapper.findAll('.ni-session-header a').at(1).trigger('click')
    expect(store.commit.mock.calls[0]).toEqual(['setModalHelp', true])
  })

  it('should close the modal on successful login', async () => {
    wrapper.setData({ fields: {
      signInPassword: '1234567890',
      signInSeed: 'bar', // <-- doesn#t check for correctness of seed
      signUpWarning: true,
      signUpBackup: true
    }})
    await wrapper.vm.onSubmit()
    expect(store.commit.mock.calls[0]).toEqual(['setModalSession', false])
  })

  it('should signal signedin state on successful login', async () => {
    wrapper.setData({ fields: {
      signInPassword: '1234567890',
      signInSeed: 'bar', // <-- doesn#t check for correctness of seed
      signUpWarning: true,
      signUpBackup: true
    }})
    await wrapper.vm.onSubmit()
    expect(store.commit.mock.calls[1][0]).toEqual('notify')
    expect(store.commit.mock.calls[1][1].title.toLowerCase()).toContain('signed up')
    expect(store.commit).toHaveBeenCalledWith('signIn', {password: '1234567890'})
  })

  it('should show error if warnings not acknowledged', () => {
    wrapper.setData({ fields: {
      signInPassword: '1234567890',
      signInSeed: 'bar',
      signUpWarning: false,
      signUpBackup: true
    }})
    wrapper.vm.onSubmit()
    expect(store.commit.mock.calls[0]).toBeUndefined()
    expect(wrapper.find('.ni-form-msg-error')).toBeDefined()
  })

  it('should show error if backup info not acknowledged', async () => {
    wrapper.setData({ fields: {
      signInPassword: '1234567890',
      signInSeed: 'bar',
      signUpWarning: true,
      signUpBackup: false
    }})
    await wrapper.vm.onSubmit()
    expect(store.commit.mock.calls[0]).toBeUndefined()
    expect(wrapper.find('.ni-form-msg-error')).toBeDefined()
  })

  it('should show error if password is not 10 long', async () => {
    wrapper.setData({ fields: {
      signInPassword: '123456789',
      signInSeed: 'bar',
      signUpWarning: true,
      signUpBackup: true
    }})
    await wrapper.vm.onSubmit()
    expect(store.commit.mock.calls[0]).toBeUndefined()
    expect(wrapper.find('.ni-form-msg-error')).toBeDefined()
  })

  it('should not continue if creation failed', async () => {
    store.dispatch = jest.fn(() => Promise.resolve(null))
    wrapper.setData({ fields: {
      signInPassword: '1234567890',
      signInSeed: 'bar',
      signUpWarning: true,
      signUpBackup: true
    }})
    await wrapper.vm.onSubmit()
    expect(store.commit).not.toHaveBeenCalled()
  })
})
