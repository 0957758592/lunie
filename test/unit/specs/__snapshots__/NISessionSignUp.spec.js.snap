// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NISessionSignUp has the expected html structure 1`] = `
"<div class=\\"ni-session\\">
   <form class=\\"ni-session-container ni-form\\">
      <div class=\\"ni-form-container\\">
         <!---->
         <main class=\\"ni-form-main\\">
            <div class=\\"ni-session-header\\">
               <a><i class=\\"material-icons\\">arrow_back</i></a>
               <div class=\\"ni-session-title\\">
                  New Account
               </div>
      <a><i class=\\"material-icons\\">help_outline</i></a>
   </div>
      <div class=\\"ni-session-main\\">
         <div class=\\"ni-form-group\\">
            <label for=\\"sign-up-seed\\">Account Seed</label>
            <div class=\\"ni-form-group-field\\">
               <textarea class=\\"ni-field\\" id=\\"sign-up-seed\\" disabled=\\"disabled\\"></textarea>
               <div class=\\"ni-form-msg ni-form-msg-desc\\">
                  Please back up the seed phrase for this account. These words cannot be recovered!
               </div>
</div>
</div>
      <div class=\\"ni-form-group\\">
         <label for=\\"sign-in-password\\">Password</label>
         <div class=\\"ni-form-group-field\\">
            <input type=\\"password\\" placeholder=\\"Password to protect your keys locally\\" class=\\"ni-field\\" id=\\"sign-in-password\\">
            <div class=\\"ni-form-msg ni-form-msg-error\\">
               Password is required
            </div>
               <!---->
   </div>
</div>
      <div class=\\"ni-form-group\\">
         <label for=\\"sign-up-warning\\"></label>
         <div class=\\"ni-form-group-field\\">
            <div class=\\"ni-field-checkbox\\">
               <div class=\\"ni-field-checkbox-input\\">
                  <input id=\\"sign-up-warning\\" type=\\"checkbox\\">
               </div>
      <label for=\\"sign-up-warning\\" class=\\"ni-field-checkbox-label\\">I understand that Cosmos cannot recover lost seed phrases.</label>
   </div>
      <div class=\\"ni-form-msg ni-form-msg-error\\">
         Recovery confirmation is required
      </div>
</div>
</div>
      <div class=\\"ni-form-group\\">
         <label for=\\"sign-up-backup\\"></label>
         <div class=\\"ni-form-group-field\\">
            <div for=\\"sign-up-backup\\" class=\\"ni-field-checkbox\\">
               <div class=\\"ni-field-checkbox-input\\">
                  <input id=\\"sign-up-backup\\" type=\\"checkbox\\">
               </div>
      <label for=\\"sign-up-backup\\" class=\\"ni-field-checkbox-label\\">I have securely backed up my generated seed phrase.</label>
   </div>
      <div class=\\"ni-form-msg ni-form-msg-error\\">
         Backup confirmation is required
      </div>
</div>
</div>
</div>
      <div class=\\"ni-session-footer\\">
         <button class=\\"ni-btn\\"><span class=\\"ni-btn-container ni-btn-size-lg\\"><i aria-hidden=\\"true\\" class=\\"ni-btn-icon material-icons\\">create</i><span class=\\"ni-btn-value\\">Create Account</span></span></button>
      </div>
</main>
      <!---->
</div>
</form>
</div>"
`;
